app/src/main/kotlin/com/hippo/ehviewer/ui/screen/DownloadsScreen.kt

```diff
@@ -128,12 +127,13 @@ import sh.calvin.reorderable.rememberReorderableLazyColumnState
 @Destination
 @Composable
 fun DownloadsScreen(navigator: DestinationsNavigator) {
-    var label by Settings.recentDownloadLabel.asMutableState()
     var gridView by Settings.gridView.asMutableState()
     var sortMode by Settings.downloadSortMode.asMutableState()
-    var keyword by rememberSaveable { mutableStateOf<String?>(null) }
-    var filterType by rememberSaveable { mutableIntStateOf(-1) }
-    var searchBarOffsetY by remember(label) { mutableIntStateOf(0) }
+    var filterState by rememberSaveable(stateSaver = DownloadsFilterState.Saver) {
+        mutableStateOf(DownloadsFilterState(Settings.recentDownloadLabel.value))
+    }
+    var invalidateKey by rememberSaveable { mutableStateOf(false) }
```

Comment:
Override `DownloadsFilterState`'s `equals` to compare instance(===), and reassign it with `filterState.copy()` directly